src/core/config/border_router.h

```diff
@@ -71,4 +71,34 @@
 #define OPENTHREAD_CONFIG_BORDER_ROUTER_REQUEST_ROUTER_ROLE 1
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE
+ *
+ * Define to 1 to enable the internal NAT64 translator.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE
+#define OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE 0
```

Comment:
We have the following:
https://github.com/openthread/openthread/blob/f790edd5e2138c662a999a755d6c1f1e6b1a0f28/src/core/config/border_routing.h#L79-L87

We should make the difference between `OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE` and `OPENTHREAD_CONFIG_BORDER_ROUTING_NAT64_ENABLE` more clear. Thoughts?

Reply:
As the scope of `NAT64_TRANSLATOR` is small enough and matches the functions, `BORDER_ROUTING_NAT64_ADVERTISEMENT` (adding "Advertisement" to distinguish it from `translator`) looks good for me, and updated the code.

One minor concern is the name might be a bit too long (`OPENTHREAD_CONFIG_BORDER_ROUTING_NAT64_ADVERTISEMENT_ENABLE`).

Reply:
Reverted, one issue it the build flag change will break the CI.

Seems we have to file an PR to ot-br-posix for the flag change first.

Reply:
Discussed with @xuyirio and decided to use NAT64_BORDER_ROUTING, this name should indicate that it is part of NAT64 functions, and it extends the existing BORDER_ROUTING module.

This change requires a change in ot-br-posix: https://github.com/openthread/ot-br-posix/pull/1505

Reply:
And there will be another PR for changing the flag.

Reply:
Sound ok to me.
If we change any existing OT config (e.g., rename it), we should add a check for it in 
`openthread-core-config-check.h`